### PR TITLE
ACNA-289 - Migrate config

### PR DESCRIPTION
**Pull request by purplecabbage**
## Description

migrate ALL data from the previous config, to the new storage

## Motivation and Context
users could be storing all sorts of data in their previous config, and they should not lose it because we changed the implementation.

## How Has This Been Tested?

Manually tested, plus updated unit tests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by purplecabbage** [comment]
#40 is related

File: src/hooks.js [hunk 1]
```diff
@@ -16,22 +17,22 @@ const toJson = (item) => {
...
 
-    let data = oldConf.get('jwt-auth')
-    oldConf.delete('jwt-auth')
-
-    if (data == null) return
-
-    oldConf.set('jwt-auth-backup', data)
-    config.set('jwt-auth', toJson(data))
-    debug('config upgraded')
+    if (data != null && Object.keys(data).length > 0) {
```
Comment (shazron) [inline src/hooks.js]: the delete above would throw if data was null. I would move the delete into the if statement

**Comment by goya** [comment]
what commit is this pull request from? i thought mine just added an “exported” flag and otherwise left everything intact.

-brett
adobe.com

________________________________

**Comment by purplecabbage** [comment]
@goya Here: https://github.com/adobe/aio-cli-plugin-config/pull/38/files#diff-95b96dc34a5bf5b035b35940dd2d2d60R32

**Comment by goya** [comment]
what other keys other than 'jwt-auth' are in legacy cli users configuration data?

**Comment by shazron** [comment]
AFAIK no other ones. All our instructions have been for modifying the `jwt-auth` key
`runtime` plugin doesn't use any config I don't think either.

**Comment by purplecabbage** [comment]
But they could have stored anything, we have no way of knowing what they put there. 

Our api was ‘aio config set x y’ so ‘aio config get x’ should always work.

**Pull request merged**